### ISSUE TITLE
is it possible to disable find references from external crates?

### ISSUE THREAD

**Issue by GopherJ** [rust-analyzer/rust-analyzer#8879]
I noticed that `find references` can be super slow when we request references in external crate, like here: if I request the references of `AllowTopLevelPaidExecutionFrom`:

<img width="785" alt="image" src="https://user-images.githubusercontent.com/33961674/118808577-3e272200-b8dc-11eb-8ced-de3ae30098db.png">

It's super slow, like more than 30seconds even one minute.

The result will be like:

<img width="1439" alt="image" src="https://user-images.githubusercontent.com/33961674/118808720-60b93b00-b8dc-11eb-9f1d-0aae8d694bb4.png">

But I'm only interested in the references located in my project, is it possible to disable `fnding references` in external crates?

**Comment by GopherJ** [comment]
<img width="773" alt="image" src="https://user-images.githubusercontent.com/33961674/118909087-0e683080-b955-11eb-86a6-f54f1f231556.png">

**Comment by GopherJ** [comment]
it seems that `find references` is extremely slow

**Comment by Veykril** [comment]
The first search will usually be very slow. I'd be nice if the language server protocol would allow the client to differ between finding all references and only references in the workspace, but since that's not a thing I guess the best option would be to offer a config setting here to toggle it, unsure.

**Comment by GopherJ** [comment]
currently RA's find references is at least 100x slower than `ripgrep`

**Comment by Veykril** [comment]
Well, ripgrep only has to find the names, not check that they are actually references to the thing you are looking for. Nevertheless there should probably be room for improvements somewhere.
Relevant issue https://github.com/rust-analyzer/rust-analyzer/issues/7404

**Comment by lnicola** [comment]
I don't think we need a setting here. A Java, C# or C++ language server won't be expected to return references from one dependency of to project to another. The fact that we have the source code and can do this doesn't mean we actually should.

**Comment by flodiebold** [comment]
I agree that we shouldn't look for references in `.cargo`, at least by default.
